hasher-matcher-actioner/hmalib/lambdas/api/datasets_api.py

```diff
@@ -0,0 +1,171 @@
+            d["write_back"],
+            d["in_use"],
+        )
+
+
+@dataclass
+class DatasetsResponse(JSONifiable):
+    datasets_response: t.List[Dataset]
+
+    def to_json(self) -> t.Dict:
+        return {
+            "datasets_response": [
+                dataset.to_json() for dataset in self.datasets_response
+            ]
+        }
+
+
+@dataclass
+class SyncDatasetResponse(JSONifiable):
+    response: str
+
+    def to_json(self) -> t.Dict:
+        return asdict(self)
+
+
+def sync_privacy_groups():
+    """
+    Implementation of the "sync_privacy_groups" function of threatexchange config.
+
+    1. call ThreatExchange API get_threat_privacy_groups_member
+    and get_threat_privacy_groups_owner to get the list of privacy groups
+
+    2. If the threat_updates_enabled is true, save it using config framework if it doesn't
+    exist in dynamoDB.
+
+    3. If one privacy group exists in dynamoDB but not in the result of get_threat_privacy_groups_owner
+    or get_threat_privacy_groups_member, update in_user to false
+    """
+    api_key = AWSSecrets.te_api_key()
+    api = ThreatExchangeAPI(api_key)
+    privacy_group_member_list = api.get_threat_privacy_groups_member()
+    privacy_group_owner_list = api.get_threat_privacy_groups_owner()
+    unique_privacy_groups = set(privacy_group_member_list + privacy_group_owner_list)
+    priavcy_group_id_in_use = set()
+
+    for privacy_group in unique_privacy_groups:
+        if privacy_group.threat_updates_enabled:
+            # HMA can only read from privacy groups that have threat_updates enabled.
+            # # See here for more details:
+            # https://developers.facebook.com/docs/threat-exchange/reference/apis/threat-updates/v9.0
+            logger.info("Adding collaboration name %s", privacy_group.name)
+            priavcy_group_id_in_use.add(privacy_group.id)
+            config = ThreatExchangeConfig(
+                privacy_group.id,
+                # TODO Currently default to True for testing purpose,
+                # need to switch it to False before v0 launch
+                fetcher_active=True,
+                privacy_group_name=privacy_group.name,
+                in_use=True,
+                write_back=False,
+            )
+            # Warning! Will stomp on existing configs (including if you disable them)
+            # TODO need to compare with existing privacy groups in dynamoDB to create/update/delete
+            hmaconfig.update_config(config, insert_only=True)
+    update_privacy_group_in_use(priavcy_group_id_in_use)
+
+
+def update_privacy_group_in_use(priavcy_group_id_in_use: set) -> None:
+    collabs = ThreatExchangeConfig.get_all()
+    for collab in collabs:
+        if collab.privacy_group_id not in priavcy_group_id_in_use:
+            hmaconfig.update_config_attributes_by_type_and_name(
+                config_type="ThreatExchangeConfig",
+                name=str(collab.privacy_group_id),
+                updates={"in_use": False},
```

Comment:
Follow up on type-safety as discussed [here](https://github.com/facebook/ThreatExchange/pull/516/files#diff-8c7b738eacee248e926da9c6d90088bc733c421aaaed4ce075c770fca8a295c7R409), a typo here "in_us" instead of "in_user" would succeed and we would never know what happened.